pages/dao/[symbol]/proposal/components/instructions/Solend/WithdrawObligationCollateralAndRedeemReserveLiquidity.tsx

```diff
@@ -0,0 +1,187 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import React, { useContext, useEffect, useState } from 'react'
+import useRealm from '@hooks/useRealm'
+import { PublicKey } from '@solana/web3.js'
+import * as yup from 'yup'
+import { isFormValid } from '@utils/formValidation'
+import {
+  UiInstruction,
+  WithdrawObligationCollateralAndRedeemReserveLiquidityForm,
+} from '@utils/uiTypes/proposalCreationTypes'
+import { NewProposalContext } from '../../../new'
+import useGovernanceAssets from '@hooks/useGovernanceAssets'
+import useWalletStore from 'stores/useWalletStore'
+import { GovernedMultiTypeAccount } from '@utils/tokens'
+import {
+  ProgramAccount,
+  serializeInstructionToBase64,
+  Governance,
+} from '@solana/spl-governance'
+import GovernedAccountSelect from '../../GovernedAccountSelect'
+import Input from '@components/inputs/Input'
+import Select from '@components/inputs/Select'
+import SolendConfiguration from '@tools/sdk/solend/configuration'
+import { withdrawObligationCollateralAndRedeemReserveLiquidity } from '@tools/sdk/solend/withdrawObligationCollateralAndRedeemReserveLiquidity'
+import BigNumber from 'bignumber.js'
+import { BN } from '@project-serum/anchor'
+
+const WithdrawObligationCollateralAndRedeemReserveLiquidity = ({
+  index,
+  governance,
+}: {
+  index: number
+  governance: ProgramAccount<Governance> | null
+}) => {
+  const connection = useWalletStore((s) => s.connection)
+  const wallet = useWalletStore((s) => s.current)
+  const { realmInfo } = useRealm()
+  const [governedAccounts, setGovernedAccounts] = useState<
+    GovernedMultiTypeAccount[]
+  >([])
+
+  const { getGovernedMultiTypeAccounts } = useGovernanceAssets()
+
+  // Hardcoded gate used to be clear about what cluster is supported for now
+  if (connection.cluster !== 'mainnet') {
+    return <>This instruction does not support {connection.cluster}</>
+  }
+
+  useEffect(() => {
+    getGovernedMultiTypeAccounts().then(setGovernedAccounts)
+  }, [])
+
+  const shouldBeGoverned = index !== 0 && governance
+  const programId: PublicKey | undefined = realmInfo?.programId
+  const [
+    form,
+    setForm,
+  ] = useState<WithdrawObligationCollateralAndRedeemReserveLiquidityForm>({
+    uiAmount: '0',
+  })
+  const [formErrors, setFormErrors] = useState({})
+  const { handleSetInstructions } = useContext(NewProposalContext)
+
+  const handleSetForm = ({ propertyName, value }) => {
+    setFormErrors({})
+    setForm({ ...form, [propertyName]: value })
+  }
+
+  const validateInstruction = async (): Promise<boolean> => {
+    const { isValid, validationErrors } = await isFormValid(schema, form)
```

Comment:
keep `isSomething` for functions that check if it's something:

`const { valid, validationErrors } = await isFormValid(schema, form)`

Also do we need to have both `valid` & `validationErrors` ? Can something be valid with errors? Can something be invalid without errors?

Reply:
copy pasted existed code, will take a deeper look

Reply:
I guess we could get rid of valid, and then just check whether validationErrors is empty or not. But it means we should also change the name of the function for something more accurate like `checkFormErrors`